### PR TITLE
Created default compare answer function

### PR DESCRIPTION
**Pull request by Daniel5055**
So I've created the default function to compare answers from a problem test case file with the result of submitted code as specified in #23. However, I am not sure how strict or lenient this comparison should be.

So far I have followed how JavaScript handles whitespace. That is, \n, \t, and ' ' are all considered whitespace (\s) and the `.trim()` function will remove those whitespaces from the start and end of strings. Furthermore, more than 1 whitespace together between non-whitespace characters will be trimmed to a single ' '. These are the only transformations I perform with strings before comparing them with JavaScripts' strict equal (===).

Examples what the function should and shouldn't compare to true are seen in the `compare.test.mjs` file.

### CODE REVIEW

File: executioner/test/compare.test.mjs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      compareAnswer(trimmable, trimmed),
+      `Comparing ${trimmable} to ${trimmed} returned false`
+    );
+  }
+});
+
+test('Comparing wrongly trimmed strings with non trimmed strings', (t) => {
+  const strings = [
+    ['Daniel\tBrathagen', 'Daniel Brathagen'],
+    ['S\tT\nA O J', 'S T A O J'],
```
Comment (DE0CH) [inline executioner/test/compare.test.mjs:89]: I think these should compare to be the same

File: executioner/src/utils/compare.mjs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+// The default function used to compare a test result with an answer
+export function compareAnswer(result, answer) {
+  function cleanString(str) {
+    // Trim spaces on end and start of string
+    const trimmedStr = str.trim();
+
+    // Replace mutiple spaces with one
+    return trimmedStr.replace(/\s\s+/g, ' ');
```
Comment (DE0CH) [inline executioner/src/utils/compare.mjs:8]: I would suggest to make this `trimdedStr.replace(/\s+/g, ' ');`

**Review by DE0CH — CHANGES_REQUESTED** [review]
I would prefer to make the function more lenient because we want to reward people for getting the right answer, not formatting it correctly. In my experience, a problem usually just asks people to output a list of numbers or a list of words, so I think replacing a newline with a space is fine.

**Comment by DE0CH** [comment]
Wait did you not agree with my reasoning or did you accidentally merged the wrong thing?

**Comment by Daniel5055** [comment]
Oh damn it, I thought I did everything right hahah. I resolved the merge conflict and merged :(

**Comment by DE0CH** [comment]
Haha that's alright. I was a bit confused at first 😂

**Comment by Daniel5055** [comment]
Damn I thought I was accepting your changes hahaha, I agree with you on replacing white space with ' ' and it does make everything feel a little more consistent also.

**Pull request merged**